server/routes/archiveGoal/index.ts

```diff
@@ -0,0 +1,27 @@
+import { Router } from 'express'
+import { Services } from '../../services'
+import ArchiveGoalController from './archiveGoalController'
+import { checkUserHasEditAuthority } from '../../middleware/roleBasedAccessControl'
+import asyncMiddleware from '../../middleware/asyncMiddleware'
+
+/**
+ * Route definitions for the pages relating to Updating A Goal
+ */
+export default (router: Router, services: Services) => {
+  const archiveGoalController = new ArchiveGoalController(services.educationAndWorkPlanService)
+
+  router.use('/plan/:prisonNumber/goals/:goalReference/archive', [checkUserHasEditAuthority()])
+  router.get('/plan/:prisonNumber/goals/:goalReference/archive', [
+    asyncMiddleware(archiveGoalController.getArchiveGoalView),
+  ])
+  router.post('/plan/:prisonNumber/goals/:goalReference/archive', [
+    asyncMiddleware(archiveGoalController.submitArchiveGoalForm),
+  ])
+  router.use('/plan/:prisonNumber/goals/:goalReference/archive/review', [checkUserHasEditAuthority()])
+  router.get('/plan/:prisonNumber/goals/:goalReference/archive/review', [
+    asyncMiddleware(archiveGoalController.getReviewArchiveGoalView),
+  ])
+  router.post('/plan/:prisonNumber/goals/:goalReference/archive/review', [
+    asyncMiddleware(archiveGoalController.submitReviewArchiveGoal),
+  ])
+}
```

Comment:
Interested in your thoughts here re: best practice / what you've done else where ....

As you've no doubt found, we have a lot of router setup functions like this one (in fact most are like this one).
IE. the router has already been defined by `app.ts`, and its passed into the function where we then add new route handlers etc to it 👍 

There is a different pattern that Lee suggested was more "standard", both in terms of express best practice, and also other HMPPS projects.
The pattern he suggested is where router functions like this do not take a router as an arg, but instead new one up internally, set it up as we are here, then return it. And the calling function (`app.ts`) uses it in an `app.use(....` construct
I think we only have 1 like this, for the new (currently unused) Notes page. See `routes/notes/index.ts`

Thoughts?
(FWIW I'm quite happy to stick to the pattern we already have and that you've used here, but I'm always open to new suggestions/ideas etc 🤔 )

Reply:
Yeah I did find this a bit clunky but it's not a big deal either way for me. Better to be consistent but maybe we do a little clean up and change them. TBH the thing I like the least is the separation of checkUserHasEditAuthority() from the controller definition, it seems like it would be easy to break in a non-obvious way.